### PR TITLE
JHtmlBehavior::formvalitator

### PR DESCRIPTION
**Pull request by degobbis**
### Summary of Changes
Some changes for use width with independend class on invalid fields.

`JHtml::_('behavior.formvalidation');` set class 'invalid' on requested empty fields
`JHtml::_('behavior.formvalidation', 'uk-form-danger');` set class ''uk-form-danger' on requested empty fields

ping @dgt41

### CODE REVIEW

**Comment by yvesh** [comment]
Ping @dgt41

File: libraries/cms/html/behavior.php [hunk 1]
```diff
@@ -188,6 +192,15 @@ public static function formvalidator()
 
 		JHtml::_('script', 'system/punycode.js', array('version' => 'auto', 'relative' => true));
 		JHtml::_('script', 'system/validate.js', array('version' => 'auto', 'relative' => true));
+
+		// Attach formvalidation to document
+		JFactory::getDocument()->addScriptDeclaration(
+			"jQuery(function() {
+						document.formvalidator = null;
+						document.formvalidator = new JFormValidator('" . $class . "');
```
Comment (yvesh) [inline libraries/cms/html/behavior.php]: Remove the initialization part from the js if you want to got this way.. Else you first load it with default settings, set it to null and load it again ;)

**Comment by dgt41** [comment]
@degobbis Please don't add a flag for the error class. You can achieve the same goal by copying the javascript files from `/media/system` to your `templates/template-name/js` and change the classes directly in the script (that's the reason Joomla has overrides)

**Comment by degobbis** [comment]
@dgt41 In my opinion the template developer is not responsible for the form validation and the normal user should not deal with it. As a developer of a form extension, I try to render the outputs for all frameworks. Because it makes sense to pass the appropriate classes for the validation. In the end, it makes the lives of the developers easier.

**Comment by mbabker** [comment]
This is supposed to be a framework agnostic helper.  So adding hooks for framework specific things seems like a step backwards.

**Comment by degobbis** [comment]
I do not understand your argument. To use a hard-coded class, which is not used in any current framework, I consider to be a step backwards. With this change I hope to integrate the same flexibility as has already happened with Modal.

**Comment by SniperSister** [comment]
I don't think that this is the right approach @degobbis - as mentioned in our Glip chat, you are trying to let extension devs fix what template devs broke. As a template dev, I should make sure that my template has the necessary stuff (=overrides or CSS) to make the generated markup look right - if I fail, it's my fault and not the extensions devs task to fix this.

**Comment by degobbis** [comment]
@SniperSister You are not wrong, but unfortunately this lack falls back on the developer. It was worth a try.

**Pull request closed without merging**